src/Tither.Domain/Pipelines/FailFastRequestBehavior.cs

```diff
@@ -0,0 +1,72 @@
+﻿
+using System.Reflection;
+using FluentValidation.Results;
+using FluentValidation;
+using MediatR;
+using OperationResult;
+using Tither.Shared.Exceptions;
+using Tither.Shared.Validators;
+
+namespace Tither.Domain.Pipelines
+{
+    public sealed class FailFastRequestBehavior<TRequest, TResponse> : IPipelineBehavior<TRequest, TResponse>
+           where TRequest : IValidatable
+    {
+        private readonly IEnumerable<IValidator> _validators;
+        private readonly MethodInfo _operationResultError;
+        private readonly Type _type = typeof(TResponse);
+        private readonly Type _typeOperationResultGeneric = typeof(Result<>);
+        private readonly Type _typeOperationResult = typeof(Result);
+
+        public FailFastRequestBehavior(IEnumerable<IValidator<TRequest>> validators)
+        {
+            _validators = validators;
+
+            if (_type.IsGenericType)
+            {
+                _operationResultError = _typeOperationResult.GetMethods().FirstOrDefault(m => m.Name == "Error" && m.IsGenericMethod);
+                _operationResultError = _operationResultError.MakeGenericMethod(_type.GetGenericArguments().First());
+            }
+        }
+
+        public Task<TResponse> Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate<TResponse> next)
+        {
+            TitherValidationException validationError;
+            List<ValidationFailure> errors;
+
+            if (_type == _typeOperationResult)
+            {
+                errors = GetErrors(request);
+                if (errors?.Any() != true)
```

Comment:
This file is a Pipelinebehavior path that checks if there is any error in the request and prevents an object from arriving at the Handler with errors.

In this specific piece of code it is checking if there are any errors and if not, it will call next?.Invoke() which will be responsible for continuing the flow.